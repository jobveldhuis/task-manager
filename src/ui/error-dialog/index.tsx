export { ErrorDialog } from "./error-dialog.component";
