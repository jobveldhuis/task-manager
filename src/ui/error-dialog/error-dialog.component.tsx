import { StyleSheet, View } from "react-native";
import { Text } from "@/ui/text";
import { Icon } from "@/ui/icon";
import { mapErrorCodeToMessage } from "@/util/map-error-code-to-message";
import { COLORS } from "@/util/colors.const";

type ErrorDialogProps = {
  code: string;
};

export function ErrorDialog({ code }: ErrorDialogProps): JSX.Element {
  const message = mapErrorCodeToMessage(code);

  return (
    <View style={styles.container}>
      <View style={styles.iconContainer}>
        <Icon name="exclamation-triangle" variant="inverse" />
      </View>
      <View style={styles.textContainer}>
        <Text variant="error">{message}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    display: "flex",
    position: "relative",
    flexDirection: "row",
    gap: 16,
    backgroundColor: COLORS.error,
    padding: 8,
    alignItems: "center",
    borderRadius: 6,
    textOverflow: "hidden",
  },
  iconContainer: {
    alignSelf: "flex-start",
    position: "relative",
    opacity: 0.5,
    top: 2,
  },
  textContainer: {
    maxWidth: "90%",
  },
});
