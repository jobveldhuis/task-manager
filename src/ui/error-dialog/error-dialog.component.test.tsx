import { ErrorDialog } from "@/ui/error-dialog/error-dialog.component";
import { render, screen } from "@testing-library/react-native";

describe("ErrorDialog", () => {
  const component = <ErrorDialog code="auth/invalid-email" />;
  it("should match snapshot", () => {
    expect(render(component).toJSON()).toMatchSnapshot();
  });

  it("should render the error message", () => {
    render(component);
    expect(
      screen.getByText("Please provide a valid e-mail address."),
    ).toBeTruthy();
  });
});
