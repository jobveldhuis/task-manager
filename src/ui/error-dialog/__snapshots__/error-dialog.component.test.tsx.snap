// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDialog should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ff2014",
      "borderRadius": 6,
      "display": "flex",
      "flexDirection": "row",
      "gap": 16,
      "padding": 8,
      "position": "relative",
      "textOverflow": "hidden",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "flex-start",
        "opacity": 0.5,
        "position": "relative",
        "top": 2,
      }
    }
  >
    <Text />
  </View>
  <View
    style={
      {
        "maxWidth": "90%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000C49",
            "fontSize": 16,
            "fontWeight": "600",
          },
          {
            "color": "#fff",
            "opacity": 0.75,
          },
          undefined,
        ]
      }
    >
      Please provide a valid e-mail address.
    </Text>
  </View>
</View>
`;
